src/class/methods.rs

```diff
@@ -157,7 +157,34 @@ pub trait PyMethodsImpl {
     fn py_methods() -> Vec<&'static PyMethodDefType> {
         inventory::iter::<Self::Methods>
             .into_iter()
-            .flat_map(PyMethodsInventory::get_methods)
+            .flat_map(PyMethodsInventory::get)
+            .collect()
+    }
+}
+
+/// Implementation detail. Only to be used through the proc macros.
+/// Similar to `PyMethodsImpl`, but for `#[pyproto]`.
+#[doc(hidden)]
+pub trait PyProtoMethodsInventory: inventory::Collect {
+    /// Create a new instance
+    fn new(methods: &'static [PyMethodDef]) -> Self;
+
+    /// Returns 'non-slot' methods for a `#[pyproto] impl` block
+    fn get(&self) -> &'static [PyMethodDef];
+}
+
+/// Implementation detail. Only to be used through the proc macros.
+/// Similar to `PyProtoMethodsImpl`, but for `#[pyproto]`.
```

Comment:
"Similar to" looks wrong as it's listing the same type?

Reply:
:+1: